core/templates/domain/statistics/LearnerActionObjectFactory.ts

```diff
@@ -20,96 +20,94 @@
-  // directly typed to 'Outcome' type once 'OutcomeObjectFactory' is upgraded.
-  actionCustomizationArgs: any;
-  schemaVersion: number;
-  /**
-   * @constructor
-   * @param {string} actionType - type of an action.
-   * @param {Object.<string, *>} actionCustomizationArgs - customization dict
-   *   for an action.
-   * @param {number} schemaVersion - schema version of the class instance.
-   */
-  // TODO(#7165): Replace 'any' with the exact type. This has been kept as
-  // 'any' because 'outcome' is an outcome domain object and this can be
-  // directly typed to 'Outcome' type once 'OutcomeObjectFactory' is upgraded.
-  constructor(
-      actionType: string, actionCustomizationArgs: any, schemaVersion: number) {
-    if (schemaVersion < 1) {
-      throw new Error('given invalid schema version');
-    }
+export interface IAnswerSubmitCustomizationArgs {
+  'state_name': {value: string};
+  'dest_state_name': {value: string};
+  'interaction_id': {value: string};
+  'submitted_answer': {value: string};
+  'feedback': {value: string};
+  'time_spent_state_in_msecs': {value: number};
+}
 
-    /** @type {string} */
-    this.actionType = actionType;
-    /** @type {Object.<string, *>} */
-    this.actionCustomizationArgs = actionCustomizationArgs;
-    /** @type {number} */
-    this.schemaVersion = schemaVersion;
-  }
+export interface IExplorationQuitCustomizationArgs {
+  'state_name': {value: string};
+  'time_spent_in_state_in_msecs': {value: number};
+}
 
-  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
-  // 'any' because the return type is a dict with underscore_cased
-  // keys which give tslint errors against underscore_casing in favor of
-  // camelCasing.
-  toBackendDict(): any {
-    return {
-      action_type: this.actionType,
-      action_customization_args: this.actionCustomizationArgs,
-      schema_version: this.schemaVersion,
-    };
-  }
+export interface ILearnerActionExplorationStart {
+  // NOTE TO DEVELOPERS: The following two definitions are NOT typos! These are
+  // discriminated unions, a feature of TypeScript. Code will fail to compile if
+  // the compiler can not deduce that action_type and schema_version are set to
+  // the following values. For details see:
+  // https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions eslint-disable max-len
+  'action_type': 'ExplorationStart';
+  'schema_version': 1;
+  'action_customization_args': IExplorationStartCustomizationArgs;
 }
 
+export interface ILearnerActionAnswerSubmit {
+  // NOTE TO DEVELOPERS: The following two definitions are NOT typos! These are
+  // discriminated unions, a feature of TypeScript. Code will fail to compile if
+  // the compiler can not deduce that action_type and schema_version are set to
+  // the following values. For details see:
+  // https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions eslint-disable max-len
```

Comment:
Please check whether eslint-disable is actually needed here and elsewhere. There is an "allow URLs" setting that may make sense to turn on, instead. /cc @Hudda 

We don't want to make disabling a norm. It should be avoided where possible.

https://eslint.org/docs/rules/max-len